packaging/docker/Dockerfile

```diff
@@ -91,6 +91,11 @@ RUN mkdir --mode=0777 data
 RUN echo 'apm-server:*:1000:1000::/usr/share/apm-server:/bin/false' >> /etc/passwd
 RUN chown -R 1000:1000 /usr/share/apm-server
 
+# remove unnecessary packages in ubuntu-based images
```

Comment:
Since https://github.com/elastic/apm-server/pull/14786, it's ubi based images. 

The command below fails when running `docker build --build-arg GOLANG_VERSION=1.23.6 -f ./packaging/docker/Dockerfile -t apm-server:test .`

Reply:
I knew I was missing something here 🤦‍♀️